### PR TITLE
feat: add countly metrics

### PR DESCRIPTION
**Pull request by SgtPooki**
Threw together a quick banner and some logic

image

### CODE REVIEW

**Comment by SgtPooki** [comment]
@juliaxbow any thoughts on simple improvements I can make to the banner without hurting the UX?

**Comment by SgtPooki** [comment]
@juliaxbow sent a mock in slack: 
image

**Comment by juliaxbow** [comment]
@whizzzkid alternative design proposal was to do a banner at the top (gdpr notice-esque). Since we have somewhat of an endless scroll, I think many users will fail to see the opt-in if it's at the bottom. Thoughts on a banner? OK with a modal box as well but this seemed less intrusive while still visible. 

Mockup (Banner):
image

In Figma: 
https://www.figma.com/file/lB6Jby1QCTLMhVG0x7TqFW/IPFS-Design-Work?node-id=2%3A660&t=23UWHv9xCE7vDOy7-0

Slack convo: 
https://filecoinproject.slack.com/archives/C03KQ8MC62Y/p1665523248891319

**Comment by SgtPooki** [comment]
> @SgtPooki are there designs available for this? I feel we're pushing down the actual content a bit too much. Maybe a modal box or how about we put this at the end of the page and just link the notices on the top?

yea I don't think we should post it at the bottom at all. it will never get accepted. It should be at the top as a minor inconvenience for those who don't choose to opt-in. I don't think it's super obtrusive at all, but I like the new mock Julia put up that reduces the vertical space it takes up

**Comment by tinytb** [comment]
Personally, I would be wary of clicking a button that says "Contribute to the improvement of IPFS" on it.

It makes me think that clicking the button will take me to GitHub where I'll directly contribute code to improve IPFS; or maybe take me a survey form where I'll have to answer a bunch of questions.

I think there is a reason that most analytics consent forms have simple, innocuous button text. Just "OK" or "Agree".

File: package.json [hunk 1]
```diff
@@ -118,6 +118,8 @@
   },
   "dependencies": {
     "@dutu/rate-limiter": "github:dutu/rate-limiter#v1.3.1",
+    "countly-sdk-web": "^22.6.2",
```
[the remark this answers is not in the document]
Comment (SgtPooki) [reply package.json:121]: I'm going to refrain from pinning without a good explicit reason to. I know ipfs-companion has some restraints that require pinning, but we don't have the same constraints here and pinning files isn't _Always a good thing_ ™

We have a lock file and that should cover our bases for the most part

**Comment by SgtPooki** [comment]
FYI @juliaxbow @whizzzkid @tinytb comments addressed and metric updates have been made

highlights:

- You can bring the banner back to adjust consent settings (necessary vs all) via a little cookie button unobtrusively above the table
- banner looks good on mobile and desktop (Julia to confirm…)
- Disagreeing still sets consent for `necessary` metrics, but displays text informing the user about this, with a link to what that means on countly.
- The app is still usable if users don't click agree or disagree, but no consent is given and thus no metrics are sent (you can confirm in network tab)
- correct consent is set when loading from localStorage (either `all` or `necessary`)

You can also test this directly on fleek now that I've got it integrated and publishing previews from PRs: https://bafybeidgyaifyhbyytawkm4lzmwwgvqlaz35h4odb5o6mspdyv4wtuxkgi.on.fleek.co/

**Comment by BigLep** [comment]
@SgtPooki : a couple of things:
1. Do you think the difference between "all" and "necessary" is clear for end users?  I know I was given links, but one was to a github issue and one is to countly page.  On the surface I wasn't clear how to compare the two.
2. What information are users going to be most concerned about that are in "all" but are not in "necessary".
3. Is there a way for a user to change this later?

**Comment by lidel** [comment]
Filled some follow-up issues:
- https://github.com/ipfs/public-gateway-checker/issues/340
- https://github.com/ipfs/public-gateway-checker/issues/342
- https://github.com/ipfs/public-gateway-checker/issues/341

**Comment by SgtPooki** [comment]
> Do you think the difference between "all" and "necessary" is clear for end users? I know I was given links, but one was to a github issue and one is to countly page. On the surface I wasn't clear how to compare the two.

Not entirely with this implementation. We will be working on addressing that, see #340 and https://github.com/ipfs/ipfs-gui/issues/125

> What information are users going to be most concerned about that are in "all" but are not in "necessary".

I will be updating https://github.com/ipfs/ipfs-gui/issues/125 to clarify the metrics that are sent with "all" 

> Is there a way for a user to change this later?

We are going to allow opt-out. further discussion happening at https://github.com/ipfs/ipfs-gui/issues/125 and https://github.com/ipfs-shipyard/ignite-metrics/issues/2

**Pull request merged**